platform/darwin/src/MGLFeature.mm

```diff
@@ -28,6 +44,17 @@ - (id)attributeForKey:(NSString *)key {
     return self.attributes[key];
 }
 
+- (NS_DICTIONARY_OF(NSString *, id) *)featureDictionary {
+    return @{@"type": @"Feature",
+             @"properties": (self.attributes) ? self.attributes : @{},
+             @"geometry": @{@"type": @"Point",
```

Comment:
I took this I idea and ran with it a bit in https://github.com/mapbox/mapbox-gl-native/pull/6524/commits/ad4e54d6a5a37bd8a589672cd35ebdd04acb711c